lightning/src/ln/channelmanager.rs

```diff
@@ -2589,6 +2594,64 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		}
 	}
 
+	// Check that an inbound payment's `payment_data` field is sane. See `create_inbound_payment` and
+	// `create_inbound_payment_for_hash` for details on our checks.
+	fn verify_inbound_payment_data(&self, payment_hash: PaymentHash, payment_data: msgs::FinalOnionHopData) -> Result<Option<PaymentPreimage>, ()> {
+		let (iv_bytes, encrypted_metadata_bytes) = payment_data.payment_secret.0.split_at(16);
+		let mut chacha = ChaCha20::new(&self.inbound_payment_key[..], &iv_bytes[..12]);
```

Comment:
Also, the `inbound_payment_key` here should be post-hkdf-extract, no?

Reply:
Fine with that. Does that add have some security advantage or is it just more conventional?

Reply:
It feels gross to hkdf-extract th key elsewhere and use it directly as the ChaCha key here. One of those things that probably doesn't matter, but is definitely not the way it "should" be done.